keras/applications/inception_resnet_v1.py

```diff
@@ -0,0 +1,379 @@
+            a common prefix `'block35_0'`.
+        activation: activation function to use at the end of the block
+            (see [activations](../activations.md)).
+            When `activation=None`, no activation is applied
+            (i.e., "linear" activation: `a(x) = x`).
+
+    # Returns
+        Output tensor for the block.
+
+    # Raises
+        ValueError: if `block_type` is not one of `'block35'`,
+            `'block17'` or `'block8'`.
+    """
+    if block_type == 'block35':
+        branch_0 = conv2d_bn(x, 32, 1)
+        branch_1 = conv2d_bn(x, 32, 1)
+        branch_1 = conv2d_bn(branch_1, 32, 3)
+        branch_2 = conv2d_bn(x, 32, 1)
+        branch_2 = conv2d_bn(branch_2, 32, 3)
+        branch_2 = conv2d_bn(branch_2, 32, 3)
+        branches = [branch_0, branch_1, branch_2]
+    elif block_type == 'block17':
+        branch_0 = conv2d_bn(x, 128, 1)
+        branch_1 = conv2d_bn(x, 128, 1)
+        branch_1 = conv2d_bn(branch_1, 128, [1, 7])
+        branch_1 = conv2d_bn(branch_1, 128, [7, 1])
+        branches = [branch_0, branch_1]
+    elif block_type == 'block8':
+        branch_0 = conv2d_bn(x, 192, 1)
+        branch_1 = conv2d_bn(x, 192, 1)
+        branch_1 = conv2d_bn(branch_1, 192, [1, 3])
+        branch_1 = conv2d_bn(branch_1, 192, [3, 1])
+        branches = [branch_0, branch_1]
+    else:
+        raise ValueError('Unknown Inception-ResNet block type. '
+                         'Expects "block35", "block17" or "block8", '
+                         'but got: ' + str(block_type))
+
+    block_name = block_type + '_' + str(block_idx)
+    channel_axis = 1 if K.image_data_format() == 'channels_first' else 3
+    mixed = Concatenate(axis=channel_axis, name=block_name + '_mixed')(branches)
+    up = conv2d_bn(mixed,
+                   K.int_shape(x)[channel_axis],
+                   1,
+                   activation=None,
+                   use_bias=True,
+                   name=block_name + '_conv')
+
+    x = Lambda(lambda inputs, scale: inputs[0] + inputs[1] * scale,
+               output_shape=K.int_shape(x)[1:],
+               arguments={'scale': scale},
+               name=block_name)([x, up])
+    if activation is not None:
+        x = Activation(activation, name=block_name + '_ac')(x)
+    return x
+
+
+def InceptionResNetV1(include_top=True,
+                      weights='facenet',
+                      input_tensor=None,
+                      input_shape=None,
+                      pooling=None,
+                      classes=44051):
+    """Instantiates the Inception-ResNet v1 architecture.
+
+    Optionally loads weights pre-trained on Facenet.
+    Note that when using TensorFlow, for best performance you should
+    set `"image_data_format": "channels_last"` in your Keras config
+    at `~/.keras/keras.json`.
+
+    The model and the weights are compatible with TensorFlow, Theano and
+    CNTK backends. The data format convention used by the model is
```

Comment:
Have you verified that this model gives similar (and reasonable) predictions on all 3 backends?